horovod/common/operations.cc

```diff
@@ -149,13 +149,16 @@ struct HorovodGlobalState {
   // Whether MPI_Init has been completed on the background thread.
   bool initialization_done = false;
 
-  // The MPI rank, local rank, size, local size and flag indicating whether MPI
-  // multi-threading is supported.
+  // The MPI rank, local rank, size, local size, flag indicating whether MPI
+  // multi-threading is supported and auto_mpi_finalize indicating whether to
+  // call MPI_Finalize automatically
   int rank = 0;
   int local_rank = 0;
   int size = 1;
   int local_size = 1;
   bool mpi_threads_supported = false;
+  bool auto_mpi_init = true;
```

Comment:
Let's rename these just `mpi_init` and `mpi_finalize` to align with parameter names.